packages/next/next-server/server/api-utils.ts

```diff
@@ -129,7 +132,7 @@ export async function parseBody(req: NextApiRequest, limit: string | number) {
  * Parse `JSON` and handles invalid `JSON` strings
  * @param str `JSON` string
  */
-function parseJson(str: string) {
+function parseJson(str: string): Object {
```

Comment:
```suggestion
function parseJson(str: string): Object {
```

`Object` is the same as saying `any` in this case.

Reply:
In some ways yes, but still is more specific than `any`

Reply:
Even do `Object` covers the majority of the cases, if the function return `null or undefined` typescript will complain

Reply:
Basically it doesn't correctly type check, see how I'm returning a string here and it passes type checking:

https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCoASMFmgY7p7ehCTkVOle4jUMdRLYTqCc8LEZzCZmoNJODPHFZZXVtZYYkAAeRJTInDQS8po+rf40gnjbDKv8LqD2jpbYoACqAEoAMsK7sUmxkGSCc+VVQQuaTwVb1UBrDYULY7PagbgUZLJH6QbYmJAECjuMigZEMVDsJzCFLNXxtajBZJcWCoCjwTrSWAACgAlPlQAB5aQAK0gNNAAG9FCUSk4CWQvABySWKAC+wQQnEENAUytAAF4xIzWcEVDgGh4JaBEPABgwQewVfdGJxmOwGMxIPxYNB1IlzfA9CIvAAiACiqwYiBq+h9SRxPvADlQkuEADlIBIfUEVBzvmRdoInVpIMdUI4FK5pHhQE4TTpRlh9CgEIgYcIrt9JKboiIDBwcd0WGwuNwgiUgA

Reply:
Yeap, `Object` type will cover string, booleans, array's, but if you return `null` or `undefined` typescript will throw an error. That's why I think is a bit more specific than `any`

Reply:
Ah damn, my bad, is not Object, is `object` on lower case. https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#number-string-boolean-symbol-and-object